nucleus/admin/rest/rest-service/src/test/java/fish/payara/admin/rest/resources/CollectionLeafResourceTest.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://github.com/payara/Payara/blob/master/LICENSE.txt
+ * See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+
+package fish.payara.admin.rest.resources;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.glassfish.admin.rest.resources.CollectionLeafResource;
+
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author asroth
+ */
+public class CollectionLeafResourceTest {
+    
+    public CollectionLeafResourceTest() {
+    }
+
+    @Test
+    public void testProcessData() {      
+        Map<String, String> data1 = new HashMap<>();
+        data1.put("-Dproduct.name=XXX", "");
+     
+        Map<String, String> data2 = new HashMap<>();
+        data2.put("-Dproduct.name", "");
+        
+        Map<String, String> data3 = new HashMap<>();
+        data3.put("-Dproduct.name=", "");
+        
+        Map<String, String> data4 = new HashMap<>();
+        data4.put("-Dproduct.name", "XXX");
+        
+        Map<String, String> data5 = new HashMap<>();
+        data5.put("-Dproduct.name", null);
+        
+        Map<String, String> data6 = new HashMap<>();
+        data6.put("-client", "");
```

Comment:
two scenarios you missed:
("-Dproduct.name=", null) looks like this would pass
("-client", null)

Reply:
Should be added now, pass as expected